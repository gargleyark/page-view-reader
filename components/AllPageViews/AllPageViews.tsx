import React from "react";
import useLogs from '../../context/LogsContext/useLogs';
import PageViewData from '../PageViewData';

const AllPageViews = (): JSX.Element => {
  const { logs } = useLogs();

  const pageViewCount = logs.reduce<Record<string, number>>((prev, { page }) => {
    if (prev[page]) {
      return { ...prev, [page]: prev[page] + 1}
    }

    return { ...prev, [page]: 1}
  }, {});

  return (
    <div>
      <h2>All page views</h2>
      {
        Object.entries(pageViewCount)
          .sort((a, b) => b[1] - a[1])
          .map(([path, count]) => <PageViewData key={path} path={path} count={count} label="page views" />)
      }
    </div>
  );
}

export default AllPageViews;