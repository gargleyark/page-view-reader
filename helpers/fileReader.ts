import { Log } from '../context/LogsContext/LogsContext';

export const formatLogs = (logs: string): Array<Log> => {
    return logs
        .split(/\n/)
        .filter(exists => !!exists)
        .map(log => ({
            page: log.split(' ')[0],
            ip: log.split(' ')[1],
        }))
}

export const getFileContent = (file: Blob) => {
    return new Promise<Array<Log>>((resolve) => {
        var fileReader = new FileReader();  
        fileReader.onloadend = () => resolve(formatLogs(`${fileReader.result}`));
        fileReader.onerror = () => resolve([]);
        fileReader.readAsText(file);
    });
};