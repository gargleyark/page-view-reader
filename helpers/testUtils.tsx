import { render } from "@testing-library/react";
import LogsContext, { Log } from '../context/LogsContext/LogsContext'

const Providers = ({ children }: { children: JSX.Element}) => {
  const logs: Array<Log> = [
    {
      page: '/home',
      ip: '1'
    },
    {
      page: '/home',
      ip: '1'
    },
    {
      page: '/about',
      ip: '1'
    },
    {
      page: '/about',
      ip: '1'
    },
    {
      page: '/about',
      ip: '2'
    },
    {
      page: '/about',
      ip: '2'
    }
  ];

  const setLogs = () => {};

  return (
    <LogsContext.Provider value={{ logs, setLogs }}>
      { children }
    </LogsContext.Provider >
  );
};

const customRender = (ui: JSX.Element, options: Record<string, string> = {}) =>
  render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";

export { customRender as render };