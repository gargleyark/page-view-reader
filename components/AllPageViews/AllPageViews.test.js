import React from "react";
import { render, screen } from "../../helpers/testUtils";
import AllPageViews from ".";

describe("All Page Views", () => {
    it('should render correctly', () => {
        const { container } = render(<AllPageViews />);
        expect(container).toMatchSnapshot();
    });
    it('should render text on page', () => {
        render(<AllPageViews />);
        expect(screen.getByText('All page views')).toBeInTheDocument();
    });
});