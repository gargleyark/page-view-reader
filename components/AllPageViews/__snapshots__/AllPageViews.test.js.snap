// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Page Views should render correctly 1`] = `
<div>
  <div>
    <h2>
      All page views
    </h2>
    <div>
      <span>
        /about
      </span>
      :
       
      <span>
        4
         
        page views
      </span>
    </div>
    <div>
      <span>
        /home
      </span>
      :
       
      <span>
        2
         
        page views
      </span>
    </div>
  </div>
</div>
`;
